matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt

```diff
@@ -1371,6 +1376,54 @@ internal class DefaultCryptoService @Inject constructor(
         }
     }
 
+    override fun sendSharedHistoryKeysToLastChunk(roomId: String, userId: String, sessionInfoSet: Set<SessionInfoPair>?) {
+        cryptoCoroutineScope.launch(coroutineDispatchers.crypto) {
+            runCatching {
+                deviceListManager.downloadKeys(listOf(userId), false)
+            }.mapCatching {
+                val userDevices = cryptoStore.getUserDevices(userId)
+                userDevices?.forEach {
```

Comment:
if one of the devices fails to export/send then the rest of the devices will be skipped, is that okay/expected?

Reply:
I'd say it's probably ok to use cache in this case. Or we should propagate exception to caller so he can do something about it.

Reply:
Can we also log something on error